src/adagrams.js

```diff
@@ -1,8 +1,133 @@
+// import { strict } from "assert";
+
 const Adagrams = {
+//WAVE 1:
   drawLetters() {
-    // Implement this method for wave 1
+// Set up hash of letters w/ frequency 
+    const letterFrequencies = {
+      'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2, 'G': 3,
+      'H': 2, 'I': 9, 'J': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8,
+      'P': 2, 'Q': 1, 'R': 6, 'S': 4, 'T': 6, 'U': 4, 'V': 2,
+      'W': 2, 'X': 1, 'Y': 2, 'Z': 1
+    };
+// WAVE 1 -->
+// use entries 
+// creating the empty lettersPool array 
+    const lettersPool = [];
+// loop through frequencies to push letters to lettersPool 
+// entries will be an array of arrays w/ letter + count in each array 
+    const entries = Object.entries(letterFrequencies);
+    entries.forEach(entry => {
+      letter = entry[0];
+      count = entry[1];
+// adding a letter into the array, w/ for loop to add the letter count number of times 
+      for (let i = 0; i < count; i += 1) {
+        lettersPool.push(letter)
+      }
+    });
+// empty array for hand of 10 random letters from lettersPool 
+    let handLetters = [];
+// for loop to draw hand of 10 random letters - ex. in animalsounds.js
+// initialization - condition - increment 
+    for (let i = 0; i < 10; i += 1) {
+// define letter to pull 10 random letters
+// pulling letters will not change lettersPool for future draws - ex. in tips.js
+//use Math.floor to round correctly and not return NaN
+      let letter = lettersPool[Math.floor(Math.random() * (lettersPool.length))];
+// push 10 selected letters into array 
+      handLetters.push(letter);
+    }
+// return array to player - explicit return 
+    return handLetters;
+  },
+// check function
+
+// WAVE 2: check if the word only uses some/all letters from handLetters array 
+// method (usesAvailableLetters w/ 2 parameters (input + lettersInHand))
+// input is a string for input word from player
+// lettersInHand is an array of 10 strings - each string is a letter 
+  usesAvailableLetters(input, lettersInHand) {
+// copying lettersInHand array so that it can be changed w/o changing original array 
+    let letters = lettersInHand;
+// initialization - condition - increment 
```

Comment:
Line 51 does not copy the array itself, it creates a new variable pointing at the same array. If you want a copy, you would need to do something like `const letters = [...lettersInHand];`.